metagpt/actions/write_code.py

```diff
@@ -135,20 +159,69 @@ async def run(self, *args, **kwargs) -> CodingContext:
         return coding_context
 
     @staticmethod
-    async def get_codes(task_doc, exclude) -> str:
+    async def get_codes(task_doc: Document, exclude: str, mode: Literal["normal", "plan"] = "normal") -> str:
+        """
+        Get code snippets based on different modes.
+
+        Attributes:
+            task_doc (Document): Document object of the task file.
+            exclude (str): Specifies the filename to be excluded from the code snippets.
+            mode (str): Specifies the mode, either "normal" or "plan" (default is "normal").
+
+        Returns:
+            str: Code snippets.
+
+        Description:
+        If mode is set to "normal", it returns code snippets for the regular coding phase,
+        i.e., all the code generated before writing the current file.
+
+        If mode is set to "plan", it returns code snippets for incremental development,
```

Comment:
why name it as "plan"? Yizhang is working wth "Plan", will the name make a confuse?